worker.go

```diff
@@ -641,3 +642,42 @@ func getCloudWatchAlarmsFromConfigMap(configMap *kubernetes.ConfigMap) aws.Cloud
 
 	return configList
 }
+
+// cniEventHandler syncronizes the events from kubernetes and the status updates from the load balancer controller.
+// Events from kubernetes pod updates trigger a rate limited update of the target groups.
+func cniEventHandler(ctx context.Context, targetCNIcfg *aws.TargetCNIconfig,
+	targetSetter func([]string, []string) error, informer func(context.Context, chan<- []string) error) {
+	var cniTargetGroupARNs []string
+	log.Infoln("Starting CNI event handler")
+
+	rateLimiter := time.NewTicker(cniEventRateLimit)
+	defer rateLimiter.Stop()
+
+	epCh := make(chan []string, 10)
+	go informer(ctx, epCh)
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case cniTargetGroupARNs = <-targetCNIcfg.TargetGroupCh:
+			log.Debugf("new message target groups: %v", cniTargetGroupARNs)
+		case eps := <-epCh:
+			log.Debugf("new message endpoints: %v", eps)
+			if len(epCh) > 0 {
+				log.Debugf("skipping, %d newer messages queued", len(epCh))
+				continue
+			}
+			if cniTargetGroupARNs == nil || len(cniTargetGroupARNs) == 0 {
```

Comment:
No need for nil check as `len(nil) == 0`

Reply:
Solely for the debug logging, then I'll move it out